src/components/JobCard/JobCard.tsx

```diff
@@ -26,21 +37,37 @@ function JobCard({
   setPendingJobs,
   filteredJobs,
   setFilteredJobs,
+  bookmarkedJobs,
+  setBookmarkedJobs,
 }: JobCardProps) {
+  const { userObject } = useContext(AuthContext);
+  const [bookmarkedValue, setBookmarked] = useState<boolean>();
+  const userObjectToString = JSON.stringify(userObject?.bookmarkedJobs);
+
+  useEffect(() => {
+    const getBookmarked = () => {
+      const bookmarks = getBookmarks(job.id, userObject);
+      setBookmarked(bookmarks);
+    };
+    getBookmarked();
+  }, [job.id, userObjectToString, userObject]);
```

Comment:
not entirely sure which of these dependencies are required for this useEffect to work properly, but i might recommend to try taking them out 1 by 1 to see if things still work. this is for cleanliness and to ensure this doesn't get triggered when we don't want it to

Reply:
if I remove job.id or userObject from the dependency array i get the following eslint error:

React Hook useEffect has a missing dependency: 'job.id'. Either include it or remove the dependency array.eslint[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)

but they both aren't technically needed